### DISCUSSION TITLE
Getting Trame to run on a Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by Marvin93S** [Kitware/trame#382] [General]
Hello,

I want to get trame running in a docker container. I followed this approach: https://github.com/Kitware/trame/tree/master/examples/deploy/docker/SingleFile

Building the image locally works good, but I am not able to deploy it on the kubernetes Cluster. I got troubles with the trame user. The kubernetes configuration does not allow non numeric users and root users. I was able to switch the user and pass the requirements of the kubernetes cluster, but then I get permission errors in the pod. What would be the best approach to solve this problem?

**Comment by jourdain** [comment]
Thanks for your feedback. If you have any ideas on how to better manage that part for kubernetes that would be great either with a PR or pointers to documentation for guidelines or recommendations. 

BTW, the trame-user has an UID/GUID of 1000.

**Comment by Marvin93S** [comment]
I still didn't get it working unfortunately. 
I am not really experienced with Kubernetes, but to me it seems like gosu is the problem here. I can see that the pod is not starting because switching the user is not permitted. I think the kubernetes configuration pod-security.kubernetes.io/enforce: restricted might be the reason. Is there a way to use trame in a restricted environment?

**Reply by jourdain** [reply]
I would think so since I know people using it within Kubernetes. But they don't rely on our docker image to do the launching and routing within the image. They do it via Kubernetes itself. 

If you rely on Kubernetes to do the scaling and networking for you, the "docker image for trame" should just be running a Python application. No Apache, no launcher, no websocket forwarding.  

With the current set of information, it is hard to provide more insight on how to move forward. But feel free to [reach out for Kitware support](https://www.kitware.com/contact) so we can adequately assess how to solve your deployment better.
